codex/stores/networkstore.nim

```diff
@@ -63,6 +64,22 @@ method putBlock*(
   await self.engine.resolveBlocks(@[blk])
   return success()
 
+method ensureExpiry*(
+    self: NetworkStore,
+    cid: Cid,
+    expiry: SecondsSince1970
+): Future[?!void] {.async.} =
+  ## Ensure that block's assosicated expiry is at least given timestamp
+  ## If the current expiry is lower then it is updated to the given one, otherwise it is left intact
+  ##
+
+  if (await self.localStore.hasBlock(cid)).tryGet:
```

Comment:
the code in the repostore can handle the case where the block is not found, right?
If yes, this method can directly pass to localstore.
If no, it probably should.